File: CS/09_Forms/RecognizeRequiredField.cs
using Spire.Pdf;
using Spire.Pdf.Fields;
using Spire.Pdf.Widget;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace RecognizeRequiredField
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
          
           // Load a PDF document
            string input = @"..\..\..\..\..\..\Data\RecognizeRequiredField.pdf";

            // Create a new instance of PdfDocument
            PdfDocument doc = new PdfDocument();

            // Load the PDF document from the specified file path
            doc.LoadFromFile(input);

            // Get the form widget from the loaded PDF document
            PdfFormWidget formWidget = doc.Form as PdfFormWidget;

            // Iterate through all the fields in the form
            for (int i = 0; i < formWidget.FieldsWidget.List.Count; i++)
            {
                // Get the current field from the FieldsWidget list
                PdfField field = formWidget.FieldsWidget.List[i] as PdfField;

                // Check if the field is required
                if (field.Required)
                {
                    MessageBox.Show("The field named: " + field.Name + " is required");
                }
            }

        }
    }
}

File: VB/09_Forms/RecognizeRequiredField.vb
Imports Spire.Pdf
Imports Spire.Pdf.Fields
Imports Spire.Pdf.Widget

Namespace RecognizeRequiredField
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Specify the input file path
			Dim input As String = "..\..\..\..\..\..\Data\RecognizeRequiredField.pdf"

			' Create a new instance of PdfDocument
			Dim doc As New PdfDocument()

			' Load the PDF document from the specified input file
			doc.LoadFromFile(input)

			' Access the form widget of the PDF document
			Dim formWidget As PdfFormWidget = TryCast(doc.Form, PdfFormWidget)

			' Iterate through each field in the form
			For i As Integer = 0 To formWidget.FieldsWidget.List.Count - 1
				Dim field As PdfField = TryCast(formWidget.FieldsWidget.List(i), PdfField)

				' Check if the field is required
				If field.Required Then
					MessageBox.Show("The field named: " & field.Name & " is required")
				End If
			Next i

			' Close the document
			doc.Close()
		End Sub
	End Class
End Namespace
